contracts/Penalizer.sol

```diff
@@ -26,8 +26,30 @@ contract Penalizer is IPenalizer{
 
     }
 
-    modifier relayManagerOnly(IRelayHub hub) {
-        require(hub.isRelayManagerStaked(msg.sender), "Unknown relay manager");
+    mapping( bytes32=>bool ) public commits;
+
+    /**
+     * any sender can call "commit(keccak(encodedPenalizeFunction))", to make sure
+     * no-one can front-run it to claim this penalization
+     */
+    function commit(bytes32 commitHash) public {
+        bytes32 idx=keccak256(abi.encodePacked(commitHash,msg.sender));
+        commits[idx] = true;
```

Comment:
I believe it would be beneficial if the hash was same for the same penalizeable event regardless of the sender. This way, this method could maybe emit the 'committed' event and then whoever wants to penalize can check is someone has already commited the same fraud proof, saving ether to potentially huge number of people.

Later, we could have some first to commit - first to get paid logic, but this is not necessary for now I think.